operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,980 @@
+                radius: self.radius,
+                stddev: self.stddev,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+pub struct GridRasterizationQueryProcessor {
+    input: TypedVectorQueryProcessor,
+    spatial_resolution: SpatialResolution,
+    grid_size_mode: GridSizeMode,
+    tiling_specification: TilingSpecification,
+    origin_coordinate: Coordinate2D,
+}
+
+#[async_trait]
+impl RasterQueryProcessor for GridRasterizationQueryProcessor {
+    type RasterType = f64;
+
+    async fn raster_query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> util::Result<BoxStream<'a, util::Result<RasterTile2D<Self::RasterType>>>> {
+        let grid_resolution = match self.grid_size_mode {
+            GridSizeMode::Fixed => SpatialResolution {
+                x: f64::max(self.spatial_resolution.x, query.spatial_resolution.x),
+                y: f64::max(self.spatial_resolution.y, query.spatial_resolution.y),
+            },
+            GridSizeMode::Relative => SpatialResolution {
+                x: f64::max(
+                    self.spatial_resolution.x * query.spatial_resolution.x,
+                    query.spatial_resolution.x,
+                ),
+                y: f64::max(
+                    self.spatial_resolution.y * query.spatial_resolution.y,
+                    query.spatial_resolution.y,
+                ),
+            },
+        };
+
+        let tiling_strategy = self
+            .tiling_specification
+            .strategy(query.spatial_resolution.x, -query.spatial_resolution.y);
+        let tile_shape = tiling_strategy.tile_size_in_pixels;
+
+        let tiles = ctx.thread_pool().install(|| {
+            stream::iter(tiling_strategy.tile_information_iterator(query.spatial_bounds)).then(
+                move |tile_info| async move {
+                    let grid_spatial_bounds = tile_info
+                        .spatial_partition()
+                        .snap_to_grid(self.origin_coordinate, grid_resolution);
+
+                    let grid_size_x =
+                        f64::ceil(grid_spatial_bounds.size_x() / grid_resolution.x) as usize;
+                    let grid_size_y =
+                        f64::ceil(grid_spatial_bounds.size_y() / grid_resolution.y) as usize;
+
+                    let vector_query = VectorQueryRectangle {
+                        spatial_bounds: grid_spatial_bounds.as_bbox(),
+                        time_interval: query.time_interval,
+                        spatial_resolution: grid_resolution,
+                    };
+
+                    let grid_geo_transform = GeoTransform::new(
+                        grid_spatial_bounds.upper_left(),
+                        grid_resolution.x,
+                        -grid_resolution.y,
+                    );
+
+                    let mut grid_data;
+
+                    if let MultiPoint(processor) = &self.input {
```

Comment:
This confused me at first.
Couldn't you let the `if let` block return the grid_data?
The else block returns anyway.

Reply:
Yes, you're right. If you want I can check whether it's MultiPoint data in the beginning and then just return a stream of empty tiles. Shouldn't really have a performance impact but probably is clearer.

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/c66df5ccc70d40de1fb304c2dba4cef57c5a3025
I also changed from returning an empty grid to a zeroed grid when the vector data type is not `MultiPoint`.
It makes more sense in this case: If there exist no points, the count is zero. Instead of the grid doesn't exist.